src/hitchhiker/tree/core.cljc

```diff
@@ -652,93 +748,97 @@ throwable error."
 (defn insert
   [{:keys [cfg] :as tree} key value]
   (go-try
-    (let [path (<? (lookup-path tree key))
-          {:keys [children] :or {children (sorted-map-by compare)}} (peek path)
-          updated-data-node (data-node cfg (assoc children key value))]
-      (loop [node updated-data-node
-             path (pop path)]
-        (if (empty? path)
-          (if (overflow? node)
-            (let [{:keys [left right median]} (split-node node)]
-              (->IndexNode [left right] (promise-chan) [] cfg))
-            node)
-          (let [index (peek path)
-                {:keys [children keys] :as parent} (peek (pop path))]
-            (if (overflow? node) ; splice the split into the parent
-              ;;TODO refactor paths to be node/index pairs or 2 vectors or something
-              (let [{:keys [left right median]} (split-node node)
-                    new-children (catvec (conj (subvec children 0 index)
-                                               left right)
-                                         (subvec children (inc index)))]
-                (recur (-> parent
-                           (assoc :children new-children)
-                           (dirty!))
-                       (pop (pop path))))
-              (recur (-> parent
-                         ;;TODO this assoc-in seems to be a bottleneck
-                         (assoc-in [:children index] node)
-                         (dirty!))
-                     (pop (pop path))))))))))
+   (let [path (<? (lookup-path tree key))
```

Comment:
Sorry about that, my emacs config re-indents code aggressively. 
Imho it would be good to do a full pass with clj-fmt on the codebase, but this is out of scope here.

Reply:
Actually `init-path` is new, we had multiple calls to (pop path) in the same iteration, ex line 666 and 676/681

Reply:
Ok, thanks.